packages/platform/atoms/cal-provider/BaseCalProvider.tsx

```diff
@@ -23,7 +30,13 @@ export function BaseCalProvider({ clientId, accessToken, options, children }: Ca
     [mutateAsync]
   );
 
-  useTimezone(handleTimezoneChange);
+  useTimezone(
+    onTimezoneChange
+      ? onTimezoneChange
+      : timezoneUpdatePreference === "auto"
+      ? handleTimezoneChange
+      : undefined
```

Comment:
Having multiple ternary operators is confusing. I think it's better to have a separate function that gets the handler and then pass it to useTimezone:

```
  const getTimezoneChangeHandler = useCallback(() => {
    if (onTimezoneChange) return onTimezoneChange;
    if (timezoneUpdatePreference === "auto") return handleTimezoneChange;
    return undefined;
  }, [onTimezoneChange, timezoneUpdatePreference, handleTimezoneChange]);

  useTimezone(getTimezoneChangeHandler());
  ```